### DISCUSSION TITLE
Accessing the correct instance in Swift Testing

### DISCUSSION THREAD

**Discussion by FabioFiorita** [hmlongco/Factory#247] [Q&A] [answered]
I'm facing an issue accessing the instance of my `AuthService` in my tests. Specifically, I can see that `MockAuthService` is being initialized, and the function is being called, but when I try to access the instance in my test using `Container.shared.authService`, it returns a different instance instead of the expected `MockAuthService`.

Here’s a simplified version of my code:

```swift
// ViewModel
@ObservationIgnored
@Injected(\.authService) private var authService

func signIn() async {
    // Sign in logic
}

// Test
@Test
func testSignInWhenSignInSucceeds() async {
    Container.shared.authService.register { MockAuthService() }
    let viewModel = SignInView.ViewModel()

    viewModel.email = "email"
    viewModel.password = "password"

    await viewModel.signIn()
    let authService = Container.shared.authService() // This returns a different instance
}

**Comment by hmlongco** [comment, marked as answer]
Yes, it does. Every time a Factory is asked to resolve an instance it returns a new instance. (Unless cached.)

**Reply by FabioFiorita** [reply]
Thank you for the clarification! I wasn't aware that it returns a new instance. I will try a different approach for my tests.
